src/main/scala/com/mwz/sonar/scala/sensor/ScalaSensor.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala.sensor
+
+import com.mwz.sonar.scala.Scala
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
+import org.sonar.api.measures.{CoreMetrics => CM}
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+/**
+ * SonarQube Sensor for the Scala programming language.
+ *
+ * @author mwz
+ * @author ElfoLiNk
+ */
+final class ScalaSensor extends Sensor {
+  override def execute(context: SensorContext): Unit = {
+    val charset = context.fileSystem().encoding.toString
+
+    val inputFiles =
+      context.fileSystem().inputFiles(context.fileSystem().predicates().hasLanguage(Scala.Key))
+
+    inputFiles.asScala.foreach { inputFile =>
+      context.newMeasure().on(inputFile).forMetric(CM.FILES).withValue(1).save()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') LiteralArguments [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.sonar-scala.com/coding_rules#rule_key=Scalastyle%3ALiteralArguments)

Reply:
Also, the *LiteralArguments* rule does not make sense when working with java code, because it is impossible to use a named argument.

Reply:
oh, that's a real shame, poor old Java 😆